spec/models/organization_spec.rb

```diff
@@ -96,33 +90,103 @@
       end
     end
   end
-  context 'search indexing' do
-    let(:org) { Fabricate :organization }
+
+  context 'searching by name' do
+    let(:gallery) { Fabricate :organization }
+    let(:zwirner) { Fabricate :organization }
+
     before do
       Organization.recreate_index!
-      org.locations.create! content: 'Berlin, Germany', city: 'Berlin', country: 'Germany'
-      tag = Fabricate :tag, name: 'fooy'
-      ot = Fabricate :organization_tag, organization: org, tag: tag
-      org.organization_tags << ot
-      org.websites.create! content: 'http://www.example.com'
-      org.organization_names.create! content: 'Quux Gallery'
-      org.save!
+
+      Fabricate(
+        :organization_name,
+        organization: gallery,
+        content: 'Quux Gallery'
+      )
+
+      Fabricate(
+        :organization_name,
+        organization: zwirner,
+        content: 'David Zwirner Gallery'
+      )
+
+      Organization.all.each(&:es_index)
+      Organization.refresh_index!
+    end
+
+    it 'includes organization names and makes them searchable' do
+      expect(Organization.estella_search(term: 'quux')).to eq([gallery])
+    end
+
+    it 'autocompletes organizations by name' do
+      expect(Organization.estella_search(term: 'quu')).to eq([gallery])
+    end
+
+    it 'uses shingle analysis' do
+      expect(Organization.estella_search(term: 'zwir')).to eq([zwirner])
+    end
+
+    it 'boosts organizations by location count' do
+      galerie = Fabricate :organization
+
+      Fabricate(
+        :organization_name,
+        organization: galerie,
+        content: 'Quux Galerie'
+      )
+
+      2.times { Fabricate :location, organization: galerie }
+
+      galerie.es_index
+      Organization.refresh_index!
+
+      expect(Organization.estella_search(term: 'quux')).to eq([galerie, gallery])
+    end
+  end
+
+  context 'searching by other organization details' do
+    let(:organization) { Fabricate :organization }
+
+    before do
+      Organization.recreate_index!
+
+      Fabricate(
+        :location,
+        organization: organization,
+        content: 'Berlin, Germany',
+        city: 'Berlin',
+        country: 'Germany'
+      )
+
+      Fabricate(
+        :organization_tag,
+        organization: organization,
+        tag: Fabricate(:tag, name: 'fooy')
+      )
+
+      Fabricate(
+        :website,
+        organization: organization,
+        content: 'http://www.example.com'
+      )
+
+      organization.es_index
```

Comment:
I don't think an explicit call is required, as it should automatically index on creation via callbacks.

Reply:
I just noticed the object is created prior to the `Organization.recreate_index!` call, so I see why you've included this now!

Reply:
❤️